src/tools/perf/uct_tests.cc

```diff
@@ -30,13 +30,18 @@ class uct_perf_test_runner {
         m_completion.func  = NULL;
 
         ucs_status_t status;
-        status = uct_iface_set_am_handler(m_perf.uct.iface, UCT_PERF_TEST_AM_ID,
-                                          am_hander, m_perf.recv_buffer, UCT_AM_CB_FLAG_DEFAULT);
+        uct_iface_attr_t attr;
+        status = uct_iface_query(m_perf.uct.iface, &attr);
         ucs_assert_always(status == UCS_OK);
+        if (attr.cap.flags & (UCT_IFACE_FLAG_AM_SHORT|UCT_IFACE_FLAG_AM_BCOPY|UCT_IFACE_FLAG_AM_ZCOPY)) {
+            status = uct_iface_set_am_handler(m_perf.uct.iface, UCT_PERF_TEST_AM_ID,
+                                              am_hander, m_perf.recv_buffer, UCT_AM_CB_FLAG_SYNC);
+            ucs_assert_always(status == UCS_OK);
+        }
     }
 
     ~uct_perf_test_runner() {
-        uct_iface_set_am_handler(m_perf.uct.iface, UCT_PERF_TEST_AM_ID, NULL, NULL, UCT_AM_CB_FLAG_DEFAULT);
+        uct_iface_set_am_handler(m_perf.uct.iface, UCT_PERF_TEST_AM_ID, NULL, NULL, UCT_AM_CB_FLAG_SYNC);
```

Comment:
probably need to add check for CB_SYNC support in libperf uct initialization part